t3/connection/src/lib.rs

```diff
@@ -0,0 +1,157 @@
+mod error;
+
+use mc_common::logger::{log, Logger};
+use mc_connection::Connection;
+use mc_util_grpc::ConnectionUriGrpcioChannel;
+use mc_util_uri::ConnectionUri;
+use t3_api::{
+    t3_grpc::TransactionServiceClient, CreateTransactionRequest, CreateTransactionResponse,
+    FindTransactionsRequest, FindTransactionsResponse, T3Uri, TestErrorRequest, TestErrorResponse,
+    TransparentTransaction,
+};
+
+use grpcio::{CallOption, ChannelBuilder, EnvBuilder, MetadataBuilder};
+
+use std::{
+    cmp::Ordering,
+    fmt::{Display, Formatter, Result as FmtResult},
+    hash::{Hash, Hasher},
+    sync::Arc,
+};
+
+pub use error::Error;
+
+pub fn common_headers_call_option(api_key: &str) -> CallOption {
+    let mut metadata_builder = MetadataBuilder::new();
+    let api_key_string = format!("ApiKey {}", api_key);
+    metadata_builder
+        .add_str("Authorization", &api_key_string)
+        .expect("Could not add api-key header");
+
+    CallOption::default().headers(metadata_builder.build())
+}
+
+#[derive(Clone)]
+pub struct T3Connection {
+    uri: T3Uri,
+    api_key: String,
+    transaction_service_client: TransactionServiceClient,
+    logger: Logger,
+}
+
+impl T3Connection {
+    pub fn new(uri: &T3Uri, api_key: String, logger: Logger) -> Self {
+        let env = Arc::new(EnvBuilder::new().name_prefix("T3RPC").build());
+        let ch = ChannelBuilder::new(env)
+            .max_receive_message_len(std::i32::MAX)
+            .max_send_message_len(std::i32::MAX)
+            .connect_to_uri(uri, &logger);
+
+        let transaction_service_client = TransactionServiceClient::new(ch);
+
+        Self {
+            uri: uri.clone(),
+            api_key,
+            transaction_service_client,
+            logger,
+        }
+    }
+
+    pub fn find_transactions(&self) -> Result<FindTransactionsResponse, Error> {
+        let request = FindTransactionsRequest::new();
```

Comment:
Seems like this should be a parameter to this method

Reply:
Ah, yeah I still need to go back and flesh out most of the t3 connection stuff, that's my immediate next task! Thanks for noting this missing :)

Reply:
That's true, although I wanted this to be a more standalone crate that isn't specific to full service, but a way for any rust app to connect to T3 in the future, so I think it's appropriate to have full functionality